packages/nodes-base/nodes/Facebook/FacebookTrigger.node.ts

```diff
@@ -148,165 +152,84 @@ export class FacebookTrigger implements INodeType {
-					headers: {
-						accept: 'application/json,text/*;q=0.99',
-					},
-					method: 'POST',
-					uri: endpoint,
-					json: true,
-					gzip: true,
-					qs: {
-						access_token: graphApiCredentials!.accessToken,
-					},
-					body: body,
-				};
+				const responseData = await facebookApiRequest.call(this, 'POST', `/${appId}/subscriptions`, body);
 
-				const webhookData = this.getWorkflowStaticData('node');
 				webhookData.verifyToken = body.verify_token;
-				const responseData = await this.helpers.request(requestOptions);
 
 				if (responseData.success !== true) {
 					// Facebook did not return success, so something went wrong
 					throw new Error('Facebook webhook creation response did not contain the expected data.');
 				}
-
-				webhookData.webhookId = uuid();
-
 				return true;
 			},
 			async delete(this: IHookFunctions): Promise<boolean> {
-				const webhookData = this.getWorkflowStaticData('node');
-
-				if (webhookData.webhookId !== undefined) {
-					const appId = this.getNodeParameter('appId') as string;
-					const graphApiCredentials = this.getCredentials('facebookGraphApi');
-
-					const endpoint = `https://graph.facebook.com/v8.0/${appId}/subscriptions`;
-					const requestOptions : OptionsWithUri = {
-						headers: {
-							accept: 'application/json,text/*;q=0.99',
-						},
-						method: 'DELETE',
-						uri: endpoint,
-						json: true,
-						gzip: true,
-						qs: {
-							access_token: graphApiCredentials!.accessToken,
-						},
-					};
-
-					try {
-						await this.helpers.request(requestOptions);
-					} catch (e) {
-						return false;
-					}
+				const appId = this.getNodeParameter('appId') as string;
+				const object = this.getNodeParameter('object') as string;
 
-					// Remove from the static workflow data so that it is clear
-					// that no webhooks are registred anymore
-					delete webhookData.webhookId;
+				try {
+					await facebookApiRequest.call(this, 'DELETE', `/${appId}/subscriptions`, { object: snakeCase(object) });
+				} catch (e) {
+					return false;
 				}
-
 				return true;
 			},
 		},
 	};
 
-
-
 	async webhook(this: IWebhookFunctions): Promise<IWebhookResponseData> {
 		const bodyData = this.getBodyData() as IDataObject;
 		const query = this.getQueryData() as IDataObject;
-
+		const res = this.getResponseObject();
 		// Check if we're getting facebook's challenge request (https://developers.facebook.com/docs/graph-api/webhooks/getting-started)
-		if (query.hasOwnProperty('hub.challenge')) {
-			const challenge = query['hub.challenge'];
-			const webhookData = this.getWorkflowStaticData('node');
-
-			if (webhookData.verifyToken !== query['hub.verify_token']) {
-				const res = this.getResponseObject();
-				res.write("Invalid verification token");
-				res.status(500).end();
-				return {};
+		if (this.getWebhookName() === 'setup') {
+			if (query['hub.challenge']) {
+				//TODO
+				//compare hub.verify_token with the saved token
```

Comment:
I tried making this work but webhookData.veryfyToken is always empty. Maybe it's an async call?
Anyway, no idea how to fix this right now.